alg_convex_hull.go

```diff
@@ -0,0 +1,119 @@
+package simplefeatures
+
+import (
+	"sort"
+)
+
+const (
+	// Right indicates the orientation is right turn which is anticlockwise
+	Right = iota
+	// Collinear indicates three points are on the same line
+	Collinear
+	// Left indicates the orientation is left turn which is clockwise
+	Left
+)
+
+// graphamScan returns a convex hull.
+func grahamScan(ps []XY) []XY {
+	if len(ps) < 3 {
+		return nil
+	}
+	sortedPoints := sortByPolarAngle(ps)
+
+	s := make([]XY, 2, len(sortedPoints))
```

Comment:
Could you name `s` as `stack` instead? If I understand correctly, it works like a stack.